src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderFooter.tsx

```diff
@@ -0,0 +1,110 @@
+import { ChevronDownIcon } from '@chakra-ui/icons'
+import {
+  Button,
+  Menu,
+  MenuButton,
+  MenuItemOption,
+  MenuList,
+  MenuOptionGroup,
+} from '@chakra-ui/react'
+import { useCallback, useMemo } from 'react'
+
+import { Row } from '@/components/Row/Row'
+import { Text } from '@/components/Text'
+import { useActions } from '@/hooks/useActions'
+import { assertUnreachable } from '@/lib/utils'
+import { ExpiryOption } from '@/state/slices/limitOrderInputSlice/constants'
+import { limitOrderInput } from '@/state/slices/limitOrderInputSlice/limitOrderInputSlice'
+import { selectExpiry } from '@/state/slices/limitOrderInputSlice/selectors'
+import { useAppSelector } from '@/state/store'
+
+const EXPIRY_OPTIONS = [
+  ExpiryOption.OneHour,
+  ExpiryOption.OneDay,
+  ExpiryOption.ThreeDays,
+  ExpiryOption.SevenDays,
+  ExpiryOption.TwentyEightDays,
+] as const
+
+const chevronDownIcon = <ChevronDownIcon />
+
+const menuButtonSx = {
+  svg: {
+    color: 'text.subtle',
+    mb: '-3px',
+  },
+  '&:hover': {
+    opacity: 0.7,
+  },
+}
+
+const getExpiryOptionTranslation = (expiryOption: ExpiryOption) => {
+  switch (expiryOption) {
+    case ExpiryOption.OneHour:
+      return `limitOrder.expiryOption.${expiryOption}`
+    case ExpiryOption.OneDay:
+      return `limitOrder.expiryOption.${expiryOption}`
+    case ExpiryOption.ThreeDays:
+      return `limitOrder.expiryOption.${expiryOption}`
+    case ExpiryOption.SevenDays:
+      return `limitOrder.expiryOption.${expiryOption}`
+    case ExpiryOption.TwentyEightDays:
+      return `limitOrder.expiryOption.${expiryOption}`
+    default:
+      assertUnreachable(expiryOption)
+  }
+}
+
+export const LimitOrderFooter = () => {
+  const expiry = useAppSelector(selectExpiry)
+  const { setExpiry } = useActions(limitOrderInput.actions)
+
+  const expiryOptions = useMemo(() => {
+    return EXPIRY_OPTIONS.map(expiryOption => {
+      return (
+        <MenuItemOption value={expiryOption} key={expiryOption}>
+          <Text translation={getExpiryOptionTranslation(expiryOption)} />
+        </MenuItemOption>
+      )
+    })
+  }, [])
+
+  const expiryOptionTranslation = useMemo(() => {
+    return getExpiryOptionTranslation(expiry)
+  }, [expiry])
+
+  const handleChangeExpiryOption = useCallback(
+    (newExpiry: string | string[]) => {
+      setExpiry(newExpiry as ExpiryOption)
```

Comment:
q: given `ExpiryOption` is an enum of string values, do we want to paranoia throw if `typeof newExpiry !== 'string'`?

Reply:
No?

MenuOptionGroup values should always be a string and it's typesafe

Reply:
<img width="1480" alt="image" src="https://github.com/user-attachments/assets/630f91f5-54e3-4e12-beef-d984b6cbb7f9" />

Reply:
Basically should look like this if we want to add it:
```typescript
diff --git a/src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderFooter.tsx b/src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderFooter.tsx
index b129b2236d..5ac3648b06 100644
--- a/src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderFooter.tsx
+++ b/src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderFooter.tsx
@@ -65,6 +65,8 @@ export const LimitOrderFooter = () => {
 
   const handleChangeExpiryOption = useCallback(
     (newExpiry: string | string[]) => {
+      if (typeof newExpiry !== 'string') throw new Error('Invalid expiry option')
+
       setExpiry(newExpiry as ExpiryOption)
     },
     [setExpiry],
```

Reply:
Added in [85910a5](https://github.com/shapeshift/web/pull/8970/commits/85910a5ad18abe38e1bd411a103dba4dffef675f)